src/server/chromium/chromium.ts

```diff
@@ -37,6 +41,34 @@ export class Chromium extends BrowserType {
       this._devtools = this._createDevTools();
   }
 
+  async cdpConnect(wsEndpoint: string, uiOptions: types.UIOptions, timeout?: number) {
+    const controller = new ProgressController();
+    controller.setLogName('browser');
+    const browserLogsCollector = new RecentLogsCollector();
+    return controller.run(async progress => {
+      const chromeTransport = await WebSocketTransport.connect(progress, wsEndpoint);
+      const browserProcess: BrowserProcess = {
+        close: async () => {
+          await chromeTransport.closeAndWait();
+        },
+        kill: async () => {
+          await chromeTransport.closeAndWait();
+        }
+      };
+      const browserOptions: BrowserOptions = {
+        ...uiOptions,
+        name: 'chromium',
+        isChromium: true,
+        headful: true,
```

Comment:
Why headful?

Reply:
To ensure we have a default context/page.

Reply:
I think we only need `persistent` for that to happen. Let's not lie to ourselves?

Reply:
Oh I thought this was in the test. Yes I have no idea why this code is here.